modules/alloy/src/test/ts/atomic/position/layout/LayoutBoundsTest.ts

```diff
@@ -0,0 +1,87 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { Arr, Optional } from '@ephox/katamari';
+import { SugarPosition } from '@ephox/sugar';
+import { assert } from 'chai';
+import * as fc from 'fast-check';
+
+import * as Boxes from 'ephox/alloy/alien/Boxes';
+import * as LayoutBounds from 'ephox/alloy/positioning/layout/LayoutBounds';
+import { assertInBounds, boundsArb } from 'ephox/alloy/test/BoundsUtils';
+
+describe('LayoutBoundsTest', () => {
+  const dimensions = [ 'left' as const, 'right' as const, 'top' as const, 'bottom' as const ];
+  const restrictionArb = boundsArb().chain<LayoutBounds.BoundsRestriction>((anchorBox) =>
+    fc.constantFrom(...dimensions).map((dim) => Arr.mapToObject(dimensions, (d): Optional<number> => {
+      if (d === dim) {
+        switch (d) {
+          case 'left':
+            return Optional.some(anchorBox.x);
+          case 'right':
+            return Optional.some(anchorBox.right);
+          case 'top':
+            return Optional.some(anchorBox.y);
+          case 'bottom':
+            return Optional.some(anchorBox.bottom);
+        }
+      } else {
+        return Optional.none();
+      }
+    }))
+  );
+
+  it('TINY-7545: the restricted layout bounds should always be inside the bounds', () => {
+    const bubbleArb = fc.integer(-20, 20);
+    fc.assert(fc.property(boundsArb(), restrictionArb, bubbleArb, bubbleArb, (bounds, restriction, bubbleLeft, bubbleRight) => {
+      const newBounds = LayoutBounds.adjustBounds(bounds, restriction, SugarPosition(bubbleLeft, bubbleRight));
+      assertInBounds(newBounds, bounds);
+    }));
+  });
+
+  it('TINY-7545: the left coord should be the anchor box/bounds plus the bubble offset', () => {
+    const bounds = Boxes.bounds(0, 0, 100, 100);
+    const inBoundsRestriction = LayoutBounds.boundsRestriction(Boxes.bounds(0, 0, 10, 100), { left: LayoutBounds.AnchorBoxBounds.RightEdge });
+    const outBoundsRestriction = LayoutBounds.boundsRestriction(Boxes.bounds(-20, 0, 10, 100), { left: LayoutBounds.AnchorBoxBounds.RightEdge });
+    fc.assert(fc.property(fc.integer(0, 90), fc.integer(0, 100), (bubbleLeft, bubbleTop) => {
```

Comment:
Can we leave something like this for when we forget why that's 90?
```suggestion
    // Make sure bubble-left is at least 10 less than the bounds so that inBoundsRestriction will always fit
    fc.assert(fc.property(fc.integer(0, 90), fc.integer(0, 100), (bubbleLeft, bubbleTop) => {
```

Reply:
How about this, as I don't like starting with `Make sure`:
> // Note: bubbleLeft needs to be at least 10 less than the bounds width to make sure it fits

Reply:
Done